buildscript-utils/src/main/kotlin/com/fueledbycaffeine/spotlight/buildscript/graph/ImplicitDependencyRule.kt

```diff
@@ -21,4 +21,20 @@ public sealed interface ImplicitDependencyRule {
     val typeSafeAccessorMap: Map<String, GradlePath>,
     override val includedProjects: Set<GradlePath> = emptySet(),
   ) : ImplicitDependencyRule
+
+  /**
+   * Gradle automatically includes intermediate parent projects of nested projects.
+   *
+   * ```
+   * :libs:foo:impl -> :libs:foo
+   * ```
+   *
+   * For Groovy, this is sort of fine because it doesn't try to compile them
+   * until they're executed. For Kotlin Gradle script, this is a problem
+   * because Gradle eagerly compiles them during configuration. So, we must
+   * treat them as implicit dependencies to work.
+   */
+  public data object KotlinGradleScriptNestingRule : ImplicitDependencyRule {
```

Comment:
Instead of making another `ImplicitDependencyRule` this could be baked in to the parsing since it is always applicable

Reply:
I thought about that and did that initially before making it configurable again since I figured groovy-only projects might want the optimization they get from not actually needing this?

Reply:
The gradle behavior applies to all projects regardless of buildscript language choice. I just didn't see the problem in any of my projects because we avoid this pattern. It has caused us issues when it sneaks into the build (which is 100% groovy scripts) in the past though!